### DISCUSSION TITLE
keep debug logging enabled

### DISCUSSION THREAD

**Discussion by ryofurue** [retorquere/zotero-better-bibtex#2400] [General]
Is there a switch to keep debug logging enabled?  I remember being provided with such a version of BBT in the past.

I seem to come across a bug whose symptom disappears when Zotero restarts.

So, currently,
Problem! ->  Try to send debug log -> Enable debug logging -> Need to restart Zotero -> Problem gone!

The problem: I create a new bib entry. BBT generates cite-key like "Author:2020". I create another entry. BBT generates a key like "Author:2020a".  I realize that the first entry was wrong and move it to the trash bin.  I ask BBT to refresh the cite-key of the second item so that it would become "Author:2020" . . . But it doesn't.

I've experienced the same problem at least several times.

**Comment by retorquere** [comment]
If you install [this plugin](https://github.com/retorquere/zotero-debug-log), it will always keep on debug logging, but you do need to restart one extra time after installation.

**Comment by ryofurue** [comment]
> If you install [this plugin](https://github.com/retorquere/zotero-debug-log)

Thanks!  Your comment made me realize
1. I already have had the plugin installed.
2. I was fooled by the panel which says "Zotero debug logging is currently disabled. It will be easier to diagnose the problem if you enable it before submitting a debug log"  with a button "Restart with Logging Enabled.."

whenever I selected "Send Better BibTeX debug log".  The fact is that the "continue" button at the bottom of this panel is blue, meaning it's clickable!

I don't know when I started to get this extra page on the debug-log panel.  When I reported problems about preprint and _eprint, I didn't see it.  For this reason, I thought that the new test builds of BBT disabled debug logging.

**Comment by retorquere** [comment]
Weird, that text should only appear when debug logging is off at the moment the popup appears.

**Comment by ryofurue** [comment]
> Weird, that text should only appear when debug logging is off at the moment the popup appears

Right.  I'm on another machine.  I just installed [the latext debug-log plugin](https://github.com/retorquere/zotero-debug-log), restarted Zotero, and tried to send a BBT debug log.  I got exactly the same text.

I then quitted Zotero again and launched it again.

Now, I don't see the dialog "Zotero debug logging is currently disabled . . . " any longer.

**Reply by retorquere** [reply]
Yeah you need an extra restart. Install + restart activates the plugin, at activation and thereafter it sets "debug logging from the start for the next restart" on. So you need one extra restart to actually have debug logging on from start.
